azure-templates/steps-pre-build.yml

```diff
@@ -91,12 +91,12 @@ steps:
   - powershell: Write-Host '##vso[task.setvariable variable=lvCLICall]LabVIEWCLI -PortNumber 3363 -LabVIEWPath "$(lvPath)\LabVIEW.exe" -AdditionalOperationDirectory "%cd%\$(buildTools)\lv\operations"'
     displayName: Set LabVIEWCLI path
 
-  - checkout: self
-    displayName: Check out repository to build
-
   - checkout: niveristand-custom-device-build-tools
     displayName: Check out NI VS Custom Device Build Tools repository
 
+  - checkout: self
+    displayName: Check out repository to build
```

Comment:
Is there a reason for reordering these? Not sure if it matters.

Reply:
Actually, what happens in the case that `self` is `niveristand-custom-device-build-tools`? Are we getting to checkouts here? Does that cause any issues?

Reply:
in the case of every other repo, it doesn't matter, but for this repo, the two checkouts will overwrite each other since they are checkouts on the same repo.  With this order, self will be checked out and will overwrite what is in build-tools main, which allows PRs to use their current repo instead of main for PR builds.

Reply:
added a comment in final commit